### PR TITLE
Feat/flocking Collision avoidance for the flocking system

### PR DESCRIPTION
**Pull request by 0x6c23**
This is an addition to the flocking/ steering systems. If flocking gets applied to an entity, it will gently (scaled by distance) steer away from obstacles. It limits clipping greatly, but it doesn't guarantee that it never happens. For that I think something else needs to be implemented. This works great, though. 

I added the NavCollisionSystem and some settings. I also renamed CollisionAvoidanceStrength to respectively ObstacleCollisionAvoidanceStrength and AgentCollisionAvoidanceStrength. The new gif reflects the new changes.

### CODE REVIEW

File: Packages/com.reese.nav/Agent/Systems/NavCollisionSystem.cs
Comment (reeseschultz) [inline Packages/com.reese.nav/Agent/Systems/NavCollisionSystem.cs]: This will get evaluated as an `int` if you use `var`, along with the `i` in the `for` loop below. Not a big deal, just letting you know.

File: Packages/com.reese.nav/Agent/Systems/NavCollisionSystem.cs
Comment (reeseschultz) [inline Packages/com.reese.nav/Agent/Systems/NavCollisionSystem.cs]: Could use a space between the `if` keyword and paren.

File: Packages/com.reese.nav/Agent/Systems/NavCollisionSystem.cs
Comment (reeseschultz) [inline Packages/com.reese.nav/Agent/Systems/NavCollisionSystem.cs]: I would rename `scaler` to `scalar`. Also, the parens are unnecessary. The division will be performed first, then the subtraction, and that's without the parens.

File: Packages/com.reese.nav/Agent/Systems/NavCollisionSystem.cs
Comment (reeseschultz) [inline Packages/com.reese.nav/Agent/Systems/NavCollisionSystem.cs:72]: There's some extra newlines here that could be removed.

File: Packages/com.reese.nav/Agent/Systems/NavCollisionSystem.cs
Comment (reeseschultz) [inline Packages/com.reese.nav/Agent/Systems/NavCollisionSystem.cs]: Maybe the `17` should be defined as a constant, and the comment could be moved above that constant.

File: Packages/com.reese.nav/Agent/Systems/NavCollisionSystem.cs
Comment (reeseschultz) [inline Packages/com.reese.nav/Agent/Systems/NavCollisionSystem.cs:25]: A newline underneath this would help with readability.

File: Packages/com.reese.nav/Flocking/NavFlockingSystem.cs
Comment (reeseschultz) [inline Packages/com.reese.nav/Flocking/NavFlockingSystem.cs]: You probably want to set this back to `false`.

**Review by reeseschultz — CHANGES_REQUESTED** [review]
Again, fantastic work. I appreciate all the effort you've put into this. The angle thing was interesting to read, in a good way.

I requested some changes, all of them are related to style except for one thing: there's a debug flag that you probably intended to be `false`, but it's `true`.

---

Aside: Speaking of the style stuff, I'll look at introducing some automated linting soon. You're only the second contributor to submit a pull request (well, two in your case), so I never gave linting any thought. The code I've written is becoming stylistically inconsistent with itself, which doesn't help, because I've been changing my mind about how it should read. Anyway, I need to do something about that with some automation beyond formatting in the text editor. **Edit:** See #72.

**Comment by reeseschultz** [comment]
@0x6c23, would you mind checking the box with the label `Allow edits from maintainers`? There are instructions on that [here](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).

I think there are some slight changes I want to make to your branch before we merge.

**Comment by 0x6c23** [comment]
It already is checked for me @reeseschultz

**Comment by reeseschultz** [comment]
I made some changes, most importantly making the obstacle steering optional via a component tag.

Thanks again, @0x6c23. You've done great work. If you plan on contributing more to this project or have any ideas you want to bounce off my head, please let me know. I can add you as a collaborator to this repository if you'd like. You can DM me on Discord via Reese#1275. Alternatively, Twitter works as well via [ReeseCodes](https://twitter.com/ReeseCodes).

**Comment by 0x6c23** [comment]
I am definitely going to contribute more, adding me as a collaborator would be great. I'll likely DM you on discord. Thanks!

**Pull request merged**
